cmake/linker/ld/target_base.cmake

```diff
@@ -11,10 +11,15 @@ macro(toolchain_ld_base)
   # TOOLCHAIN_LD_FLAGS comes from compiler/gcc/target.cmake
   # LINKERFLAGPREFIX comes from linker/ld/target.cmake
   zephyr_ld_options(
-    -no-pie
     ${TOOLCHAIN_LD_FLAGS}
   )
 
+  if(CONFIG_BOARD_NATIVE_POSIX)
+    zephyr_ld_options(
+      ${LINKERFLAGPREFIX},-no-pie
```

Comment:
your commit message states:
> Zephyr SDK and others are, currently, not to produce PIE binaries by default.

I get that if Zephyr SDK and other currently do not produce PIE binaries per default, but what is the problem in specifying the flag in those cases then ?

Also, we are explicitly requesting the compiler to not do position independent code for `C` code here:
https://github.com/zephyrproject-rtos/zephyr/blob/ab764aada99a420ef7fe75ec6ea8984391fae2a5/CMakeLists.txt#L376-L377

so having `-no-pie` for the linker seems logical to me in this case, even though the toolchain you mention does so per default.

Is your intention to be able to actually use `-pie` on the linker in some use-cases ?

Or should we actually change the handling of `pie` / `-no-pie`, so that it compiler and linker flags are set as properties and then applied based on configuration.
Similar to how things are done for `coverage`, where the are both compiler and linker flags specified under the `coverage` property, which is then applied based on Kconfig.
Ref:
https://github.com/zephyrproject-rtos/zephyr/blob/d598e26de42cf9fc8ba4e9d048db33e0a82004bf/arch/common/CMakeLists.txt#L72-L75

Reply:
> so having `-no-pie` for the linker seems logical to me in this case, even though the toolchain you mention does so per default.
> 
> Is your intention to be able to actually use `-pie` on the linker in some use-cases ?

Answering my own question.
I see the need for not passing the flag here: https://github.com/zephyrproject-rtos/zephyr/pull/54333#issue-1566858160

So atm it seems to be specific to the Xtensa toolchain, `xcc`.
Which means the best approach forward is to have a `no_position_independent` property also on the linker, and the Xtensa toolchain can then set this property flag to empty in `https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/linker/ld/xcc/linker_flags.cmake` (assuming the xcc uses the ld linker)

Reply:
I believe we need this with LLVM as well.  So it might not be xcc specific.